auto_version

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+function git_current_branch() {
+    git symbolic-ref --short HEAD 2>/dev/null; or git rev-parse --short HEAD 2>/dev/null
+}
+
+current_branch=$(git_current_branch)
+
+if [[ ! $current_branch =~ ^(master|main)$ ]]; then
+  echo "this script can only work with branch master or main"
+  exit 255
+fi
+
+# update before tagging
+git fetch --all
+for remote in $(git remote); do
+  git pull "$remote" "$current_branch"
+done
+
+for pkg in $(find . -name 'go.mod' -exec dirname {} \; | sort -u); do
+  pkg=$(echo "$pkg" | sed 's/\.\///')
+  latest_version=$(git tag --sort=committerdate -l "$pkg/*" | head -1 | grep -Eo 'v.*')
+  minor_version=$(echo "$latest_version" | rev | cut -d'.' -f 1 | rev)
```

Comment:
This should be the `patch`, not `minor` according to the [semantic version](https://semver.org/).
Shall we add the options for this script to allow us to:
- Tag new `patch` version
- Tag new `minor` version
- Tag new `major` version

Reply:
Actually here just update the `minor` version here `echo "$latest_version" | rev | cut -d'.' -f 1 | rev`
I haven't tried to update `major` and `path`.
Let me try to do the remaings

Reply:
@mysqto 
Yes, it's nice to support all `major.minor.patch`
And `echo "$latest_version" | rev | cut -d'.' -f 1` will be enough to get the `patch` (last number), don't need to have the last `rev`.

Reply:
@viettd  if we don't rev it back, `13` will be `31`

Reply:
You're right, I didn't think about that 😂

Reply:
Actually an easier way is that we assume the version number will always be `path.major.minor`, then we can use `f1` as `path`, `f2` as `major`, `f3` as `minor`. I use `rev` here, just in case that we don't follow the rule and just get the last filed when cutting with `.`

Reply:
let's make it simple 😂, we always follow the rules 😂